eZ/Publish/Core/Limitation/LanguageLimitationType.php

```diff
@@ -85,66 +126,186 @@ public function validate(APILimitationValue $limitationValue)
     /**
      * Create the Limitation Value.
      *
-     * @param mixed[] $limitationValues
+     * @param array[] $limitationValues
      *
      * @return \eZ\Publish\API\Repository\Values\User\Limitation
      */
-    public function buildValue(array $limitationValues)
+    public function buildValue(array $limitationValues): APILimitationValue
     {
-        return new APILanguageLimitation(array('limitationValues' => $limitationValues));
+        return new APILanguageLimitation(['limitationValues' => $limitationValues]);
     }
 
     /**
-     * Evaluate permission against content & target(placement/parent/assignment).
-     *
-     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException If any of the arguments are invalid
-     *         Example: If LimitationValue is instance of ContentTypeLimitationValue, and Type is SectionLimitationType.
-     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException If value of the LimitationValue is unsupported
-     *         Example if OwnerLimitationValue->limitationValues[0] is not one of: [ 1,  2 ]
+     * Evaluate permission against content & target.
      *
-     * @param \eZ\Publish\API\Repository\Values\User\Limitation $value
-     * @param \eZ\Publish\API\Repository\Values\User\UserReference $currentUser
+     * {@inheritdoc}
+     */
+    public function evaluate(
+        APILimitationValue $value,
+        APIUserReference $currentUser,
+        ValueObject $object,
+        array $targets = null
+    ): ?bool {
```

Comment:
I think I am missing the point where null could be returned

Reply:
> I think I am missing the point where null could be returned

It needs to be compatible with `\eZ\Publish\SPI\Limitation\Type::evaluate`.

We have `ACCES_GRANTED`, `ACCESS_DENIED` and the `ACCESS_ABSTAIN` which is `null`.
One of the future ideas is to follow Symfony closer and return 1, 0, -1 respectively, so there will be no confusion.